### ISSUE TITLE
column selection via index 

### ISSUE THREAD

**Issue by lauritzthamsen** [citlab/dstat-tools#36]
it would be great if the plot-script would also be usable for dstat recordings without headers. we have this case when dstat is started and stopped for entire peel _experiments_, yet for _experiment runs_ only the rows are copied that were added during the run (i.e. no headers). 

so instead of writing

> ruby dstat_plot.rb -c "total cpu usage" -f "usr" example.csv

i'd like to execute something like

> ruby dstat_plot.rb -col 6 example.csv

**Comment by joh-mue** [comment]
What should the files be named then? the name used to be based on the specified category and field. Now it defaults to "column X.png". I could give the option to specifiy the name with --name NAME

**Comment by lauritzthamsen** [comment]
yeah, maybe "dstat_colX" instead of just "column X". and yes, having the option to provide the name would be nice too.

**Comment by lauritzthamsen** [comment]
@verbit with `--name` we could maybe add some infos on the peel experiment / suite to the filename

**Comment by verbit** [comment]
The filename is already configurable via #35 (option `-o | --output`).

**Comment by joh-mue** [comment]
@verbit oops. this is correct. did not see that.

**Comment by lauritzthamsen** [comment]
what's the status here?

**Comment by joh-mue** [comment]
branch https://github.com/citlab/dstat-tools/tree/column_rework provides the functionality.
do you want to try it from the branch first and see if it works properly? Otherwise merge pull request https://github.com/citlab/dstat-tools/pull/37

**Comment by verbit** [comment]
@joh-mue Can you please update the `README` to reflect the recent changes? An example of column selection via index would be nice :)

**Comment by trenner** [comment]
@joh-mue indexing with -l does not work for dstats records with dstat headers... Any idea for fixing it?

dstat_plot.rb:90:in `transpose': element size differs (7 should be 0) (IndexError)
    from dstat_plot.rb:90:in`block in read_column_from_csv'
    from dstat_plot.rb:77:in `each'
    from dstat_plot.rb:77:in`read_column_from_csv'
    from dstat_plot.rb:303:in `<main>'

**Comment by joh-mue** [comment]
@verbit will do.

@trenner it actually should just skip the header in that case but I will check

**Comment by joh-mue** [comment]
@trenner what does your header look like? Can you actually send me a current example of a dstat log?

**Comment by trenner** [comment]
thomas@thomas-cit:~/Work/gitlab/dstat-tools/data$ cat tpch3.scale-out.nodes-21/tpch3.flink.scale-out.nodes.top004.con4.rep1.run01/logs/dstat/dstat-0.7.2/dstat-trenner-wally003.csv


"Host:","wally003.cit.tu-berlin.de",,,,"User:","trenner"
"Cmdline:","dstat --epoch --cpu -C total,1,2,3,4,5,6,7,8 --mem --net -N total,eno1,enp4s0 --disk -D total,sda,sdb,sdc,sdd,sde,sdf,sdg,sdh --noheaders --nocolor --output /home/trenner/peel-experiments/ioaware-scheduling/systems/dstat-0.7.2/log/dstat-trenner-wally003.csv 1",,,,"Date:","20 Jan 2016 18:28:59 CET"

"epoch","total cpu usage",,,,,,"cpu1 usage",,,,,,"cpu2 usage",,,,,,"cpu3 usage",,,,,,"cpu4 usage",,,,,,"cpu5 usage",,,,,,"cpu6 usage",,,,,,"cpu7 usage",,,,,,"memory usage",,,,"net/total",,"net/enp4s0",,"dsk/total",,"dsk/sda",,"dsk/sdb",,"dsk/sdc",,"dsk/sdd",,"dsk/sde",,"dsk/sdf",,"dsk/sdg",,"dsk/sdh",
"epoch","usr","sys","idl","wai","hiq","siq","usr","sys","idl","wai","hiq","siq","usr","sys","idl","wai","hiq","siq","usr","sys","idl","wai","hiq","siq","usr","sys","idl","wai","hiq","siq","usr","sys","idl","wai","hiq","siq","usr","sys","idl","wai","hiq","siq","usr","sys","idl","wai","hiq","siq","used","buff","cach","free","recv","send","recv","send","read","writ","read","writ","read","writ","read","writ","read","writ","read","writ","read","writ","read","writ","read","writ"
1453310939.195,0.804,0.103,99.068,0.023,0.000,0.003,0.773,0.057,99.157,0.013,0.0,0.000,0.789,0.062,99.147,0.002,0.0,0.000,0.778,0.060,99.160,0.001,0.000,0.000,0.772,0.122,98.946,0.159,0.000,0.001,0.986,0.221,98.769,0.004,0.000,0.019,0.780,0.118,99.098,0.002,0.000,0.003,0.775,0.121,99.103,0.001,0.0,0.000,2465144832.0,0.0,12026834944.0,2138710016.0,0.0,0.0,0.0,0.0,141820.057,283752.645,30622.577,94601.818,30633.121,94601.390,30622.755,94549.434,0.078,0.0,49803.101,0.002,0.028,0.0,138.371,0.0,0.027,0.0

**Comment by joh-mue** [comment]
okay. of course yours is slightly different. I was expecting

```
"Dstat 0.7.2 CSV output"
"Author:","Dag Wieers <dag@wieers.com>",,,,"URL:","http://dag.wieers.com/home-made/dstat/"
```

at the very top. I will adapt the script accordingly.

**Comment by trenner** [comment]
Hi, I think it just happens with the peel extension. The log starts with 2 empty lines and may skips the lines you mentioned.

[dstat-trenner-wally002.txt](https://github.com/citlab/dstat-tools/files/100491/dstat-trenner-wally002.txt)

**Comment by trenner** [comment]
if it is straight forward, please adapt your script. But I think the main problem is that the dstat peel-extension starts too late somehow and scripts some lines . @lauritzthamsen @akunft have similar problems. If we fix this, there is no need for your fix... :) So please don't spend to much time on it.

**Comment by aalexandrov** [comment]
We should get this done once we merge stratosphere/peel#71

**Comment by verbit** [comment]
Maybe I can give you some insights on this problem.

The two missed lines are there intentionally. What happens is that dstat recognizes if it outputs to a file into which it has already written and therefore skips the first two headers (since they are already there at the beginning).

This happens since the log files are kept between the individual lifecycles of the peel-extension. The fix should include checking for old log files and renaming them if necessary (for example to `.csv.1`) in the `setUp` method.

**Comment by joh-mue** [comment]
thanks @verbit for the clarification. Is my understanding correct that dstat will write the first two header lines only once and a set of [headers + values] for each consecutive run so that there would be headers throughout the csv file?

If there is only 3 cases: full header, short header, no header - that wouldn't be too hard to fix actually, since the first and the last case are already in there. Having new sets of headers throughout a file sounds very annoying to deal with. It also is not conforming with [RFC 4180](https://www.ietf.org/rfc/rfc4180.txt). ;)
In that case having separated, consecutively numbered files would be the preferable options I guess

**Comment by verbit** [comment]
> Is my understanding correct that dstat will write the first two header lines only once and a set of [headers + values] for each consecutive run so that there would be headers throughout the csv file?

Correct, however, the `dstat_plot.rb` is not dealing with the original `csv` file. Instead, `dstat_plot.rb` works on cuttings from the original file which correspond to exactly one run. This means that the tool gets files with only one header as input (just that the header can have 2 forms, full or short).

When this is fixed in the peel extension (as suggested above), the tool should always get files with a full header.

**Comment by joh-mue** [comment]
okay it's fixed on the master. I'm making the assumption that in case of the short header the first two lines are still present but empty.

I also updated the readme, put in a Column selection example after the graphic and put column numbers in the table at the bottom.

**Comment by joh-mue** [comment]
implemented with PR #37 - `read from column finished`
